src/Ocean/CartesianDomains/cartesian_field.jl

```diff
@@ -0,0 +1,109 @@
+using ClimateMachine.MPIStateArrays: MPIStateArray
+
+using ..Fields: AbstractField
+
+import ..Fields: field
+
+#####
+##### CartesianField
+#####
+
+struct CartesianField{E, D} <: AbstractField
+    elements::E
+    domain::D
+end
+
+function Base.collect(field::CartesianField)
+    elements = [
+        collect(field.elements[i, j, k])
+        for
+        i in 1:(field.domain.Ne.x),
+        j in 1:(field.domain.Ne.y), k in 1:(field.domain.Ne.z)
+    ]
+    return CartesianField(elements, field)
+end
+
+Base.maximum(f, field::CartesianField) =
+    maximum([maximum(f, el) for el in field.elements])
+Base.minimum(f, field::CartesianField) =
+    minimum([minimum(f, el) for el in field.elements])
+
+Base.maximum(field::CartesianField) =
+    maximum([maximum(el) for el in field.elements])
+Base.minimum(field::CartesianField) =
+    minimum([minimum(el) for el in field.elements])
+
+Base.show(io::IO, field::CartesianField{FT}) where {FT} =
+    print(io, "CartesianField{$FT}")
+
+Base.@propagate_inbounds Base.getindex(field::CartesianField, i, j, k) =
+    field.elements[i, j, k]
+
+Base.size(field::CartesianField) = size(field.elements)
+
+"""
+    CartesianField(domain::CartesianDomain, state::MPIStateArray, variable_index::Int)
+
+Returns an abstracted Cartesian `view` into `state.realdata[:, variable_index, :]`
+that assumes `state.realdata` lives on `CartesianDomain`.
+
+`CartesianField.elements` is a three-dimensional array of `RectangularElements`.
+"""
+function CartesianField(
+    domain::CartesianDomain,
+    state::MPIStateArray,
+    variable_index::Int,
+)
+
+    # Unwind the data in solver
+    data = view(state.realdata, :, variable_index, :)
+
+    # Unwind volume geometry
+    volume_geometry = domain.grid.vgeo
+
+    Ne = domain.Ne
+    Te = prod(domain.Ne)
+    Np = domain.Np
+
+    # Extract coordinate arrays with size (xnode, ynode, znode, element)
+    x = reshape(volume_geometry[:, 13, :], Np + 1, Np + 1, Np + 1, Te)
+    y = reshape(volume_geometry[:, 14, :], Np + 1, Np + 1, Np + 1, Te)
+    z = reshape(volume_geometry[:, 15, :], Np + 1, Np + 1, Np + 1, Te)
```

Comment:
Can we grab these indexes from `Grids`?

Reply:
@charleskawczynski and @jkozdon do you have an opinion which is the authoritative variable out of ```_x1``` and ```x1id```?  ```src/Numerics/Mesh/Grids.jl``` has a loop for ```_``` prefix things and then defines some suffix ```id``` variables . I think an originalist view could be that the framers of the original code meant x1id to be authoritative. But _x1 and x1id seem to get used somewhat equally in under more recent rulings.  The original framers in their wisdom do not seem to have left too much guidance in their writings that I could find. 

```
const _nvgeo = 16
const _ξ1x1,
_ξ2x1,
_ξ3x1,
_ξ1x2,
_ξ2x2,
_ξ3x2,
_ξ1x3,
_ξ2x3,
_ξ3x3,
_M,
_MI,
_MH,
_x1,
_x2,
_x3,
_JcV = 1:_nvgeo
const vgeoid = (
    ξ1x1id = _ξ1x1,
    ξ2x1id = _ξ2x1,
    ξ3x1id = _ξ3x1,
    ξ1x2id = _ξ1x2,
    ξ2x2id = _ξ2x2,
    ξ3x2id = _ξ3x2,
    ξ1x3id = _ξ1x3,
    ξ2x3id = _ξ2x3,
    ξ3x3id = _ξ3x3,
    Mid = _M,
    MIid = _MI,
    MHid = _MH,
    x1id = _x1,
    x2id = _x2,
    x3id = _x3,
    JcVid = _JcV,
)
```

test/Numerics/DGMethods/horizontal_integral_test.jl uses ClimateMachine.Mesh.Grids.vgeoid.x1id.

test/Diagnostics/diagnostic_fields_test.jl uses import ClimateMachine.Mesh.Grids: _x1, _x2, _x3

Reply:
I'm not sure. @jkozdon?

Reply:
If I may, I would prefer to raise an issue to address this, given its experimental nature. I am going to submit a PR soon that may render this function obsolete.

Reply:
I made the requisite changes to resolve this conversation, but I'll wait for @jkozdon's input.

Reply:
I think that we mostly use the underscore versions in the kernels.

A lot of this code was written when @lcw and I were just learning Julia, trying out different ideas, etc.. I don't remember for sure, but maybe we originally planned on using the `vgeoid` everywhere, but this didn't work with our initial GPU code so we fell back to the underscore. But I really don't know for sure!

We should probably just use one and delete the other.

Reply:
Sounds like cleaning up the ambiguity between `x1id` and `_x1` deserves a new issue? Since I addressed the original question I'm going to resolve this conversation.